dashboard/app/helpers/milestone_helper.rb

```diff
@@ -0,0 +1,83 @@
+module MilestoneHelper
+  include ScriptLevelsHelper
+  include LevelsHelper
+
+  def milestone_response(options)
```

Comment:
It would be good to push as much of this logic as possible into the `Activity` model (or non AR-backed Ruby object that lives in the /models directory).  Can the test coverage be updated to run against this method directly?